lib/spack/spack/util/ordereddict.py

```diff
@@ -0,0 +1,37 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+"""This file dispatches to the correct implementation of OrderedDict."""
+
+# TODO: this file can be removed when support for python 2.6 will be dropped
+
+import sys
+
+if sys.version_info[:2] == (2, 6):
+    # This hack is needed because otherwise we'll load
+    # spack.util.ordereddict for some reason.
```

Comment:
"For some reason" makes me nervous. I'd be interested to investigate why. What "should" have worked instead?

Reply:
I think one of the sources of complexity here is that the file is named `ordereddict.py`. My understanding is that this file was created in the first place to avoid introducing conditional import logic wherever `OrderedDict` is used. IMO things would be more simple if this were named `backports.py`. I'll attach a patch which updates your PR with that suggestion.

Reply:
Gist here: https://gist.github.com/scheibelp/5fd82f5fbdb670f72883772f1c1a6965

Reply:
@scheibelp @adamjstewart 
> "For some reason" makes me nervous.

Ok, here's the reason: [PEP-328](https://www.python.org/dev/peps/pep-0328/). It seems the absolute imports were not mandatory for that version of python. I solved this issue using:
```python
from __future__ import absolute_import
```
and didn't rename the module in `spack.util`. If you still prefer `backports` let me know and I'll do the renaming.